src/platform/backends/shared/linux/apparmor.cpp

```diff
@@ -48,7 +49,7 @@ void throw_if_binary_fails(const char* binary_name, const QStringList& arguments
 
 QStringList generate_extra_apparmor_args()
 {
-    if (mp::utils::is_snap())
+    try
```

Comment:
Two questions about making apparmor dependent on the environment vars. I expect we are safe on both, but just to be absolutely sure:
  - are the 3 vars guaranteed to be defined when in a strict snap? is there any known way to unset/overwrite them? (it would be a preexisting issue, but a pressing one)
  - would `SNAP_NAME` or `SNAP_COMMON` be any easier to unset/overwrite than `SNAP` - if so, that would be an issue with this PR.

Reply:
> are the 3 vars guaranteed to be defined when in a strict snap?

Yes, and not just strict, but classic as well.

> is there any known way to unset/overwrite them?

Since the snap vars are set by snapd and *only* seen by the running snap, the only way would be for the snap itself to change them.  If something _outside_ of the snap can unset/overwrite them, then that is a serious failing of the whole snap architecture.

> would SNAP_NAME or SNAP_COMMON be any easier to unset/overwrite than SNAP - if so, that would be an issue with this PR.

See my answer above.  Short answer, it would be a serious failing of the whole snap architecture.

Reply:
OK, good to know.